astropy/cosmology/parameter/_core.py

```diff
@@ -43,68 +97,64 @@ class Parameter:
     For worked examples see :class:`~astropy.cosmology.FLRW`.
     """
 
-    def __init__(
-        self,
-        *,
-        derived=False,
-        unit=None,
-        equivalencies=[],
-        fvalidate="default",
-        doc=None,
-    ):
-        # attribute name on container cosmology class.
-        # really set in __set_name__, but if Parameter is not init'ed as a
-        # descriptor this ensures that the attributes exist.
-        self._attr_name = self._attr_name_private = None
+    if not PYTHON_LT_3_10:
+        _: KW_ONLY
 
-        self._derived = derived
-        self.__doc__ = doc
+    derived: bool = False
+    """Whether the Parameter is derived; true parameters are not."""
 
-        # units stuff
-        self._unit = u.Unit(unit) if unit is not None else None
-        self._equivalencies = equivalencies
+    # Units
+    unit: _UnitField = _UnitField()  # noqa: RUF009
 
-        # Parse registered `fvalidate`
-        self._fvalidate_in = fvalidate  # Always store input fvalidate.
-        if callable(fvalidate):
-            pass
-        elif fvalidate in _REGISTRY_FVALIDATORS:
-            fvalidate = _REGISTRY_FVALIDATORS[fvalidate]
-        elif isinstance(fvalidate, str):
-            raise ValueError(
-                f"`fvalidate`, if str, must be in {_REGISTRY_FVALIDATORS.keys()}"
-            )
-        else:
-            raise TypeError(
-                f"`fvalidate` must be a function or {_REGISTRY_FVALIDATORS.keys()}"
-            )
-        self._fvalidate = fvalidate
+    equivalencies: u.Equivalency | Sequence[u.Equivalency] = field(default_factory=list)
+    """Equivalencies used when initializing Parameter."""
```

Comment:
Is this supposed to be a string literal instead of a comment?

Reply:
Sphinx can (/should) pick up on comments directly after fields and display it in the docs.
In my experience the mechanism is flaky.

Reply:
Sphinx seems to indeed pick up many of these ([link to `Parameter` docs](https://astropy--14874.org.readthedocs.build/en/14874/api/astropy.cosmology.Parameter.html#astropy.cosmology.Parameter)), but I don't see any documentation for `equivalencies`. Is that how it is supposed to be?

Reply:
If it's not picking it up, I'll remove the comment. I think the field is largely self explanatory, so not work the effort wrangling Sphinx.

Reply:
Done!

Reply:
Sphinx did recognize the docstrings for what is listed under the Attributes Summary and those docstrings were nice to have. But for some reason `equivalencies` is not listed among the attributes at all.

Reply:
Strange, when I build the docs myself it works great. Trying again with RTD.

<img width="639" alt="CleanShot 2023-07-28 at 16 39 57@2x" src="https://github.com/astropy/astropy/assets/8949649/1a49e909-b1a0-466c-9d84-6a2fd640b2f6">

Reply:
It works on RTD now!